### PR TITLE
feat: support for elasticsearch 8.7

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

* Supporting Elasticsearch 8.7 version.
* Removed EOL versions 7.9 and 7.10
* Updated tests accordingly.

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: version.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.3.0"
+  required_version = "<= 1.5.5"
```
Comment (ocofaigh) [inline version.tf:2]: `>= 1.3.0, <= 1.5.5`

File: examples/complete/version.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.3.0"
+  required_version = "<= 1.5.5"
```
Comment (ocofaigh) [inline examples/complete/version.tf:2]: `>= 1.3.0, <= 1.5.5`

File: examples/basic/version.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.0.0"
+  required_version = "<= 1.5.5"
```
Comment (ocofaigh) [inline examples/basic/version.tf:2]: `>= 1.3.0, <= 1.5.5`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

File: tests/other_test.go [hunk 4]
```diff
@@ -18,7 +18,7 @@ func TestRunBasicExample(t *testing.T) {
 		ResourceGroup:      resourceGroup,
 		BestRegionYAMLPath: regionSelectionPath,
 		TerraformVars: map[string]interface{}{
-			"elasticsearch_version": "7.10", // test the version that is not currently tested in pr_test.go
+			"elasticsearch_version": "8.7", // test latest or untested version of the Elasticsearch
```
[the remark this answers is not in the document]
Comment (rajatagarwal-ibm) [reply tests/other_test.go:21]: sounds good.

[the remark this answers is not in the document]
Comment (rajatagarwal-ibm) [reply tests/other_test.go:21]: Perhaps it will take the default value from example/main

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Hey @rajatagarwal-ibm sorry, turns out the BSL license is only coming in tf 1.6, so plese update the range to be `>= 1.3.0, < 1.6.0`

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**